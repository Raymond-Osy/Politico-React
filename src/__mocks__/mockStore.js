export const mockStoreData = {
  auth: {
    signupLoading: false,
    signupSuccess: false,
    singupFailure: false,
    signupError: {},
    loginLoading: false,
    loginSuccess: false,
    loginFailure: false,
    loginError: {}
  },
  parties: {
    partiesRequest: false,
    partiesSuccess: false,
    partiesFailure: false,
    partiesError: {},
    parties: []
  },
  toastr: {}
};
