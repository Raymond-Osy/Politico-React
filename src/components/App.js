import React, { Component } from 'react';

class App extends Component {
    componentDidMount(){
        console.log('I am a chosen');
    }

    render() {
        return (
            <div>
                <h1>This is my Politico react app</h1>
            </div>
        );
    }
}

export default App;