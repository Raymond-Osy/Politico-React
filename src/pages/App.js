import React from 'react';
import {
  BrowserRouter as Router, Switch, Route
} from 'react-router-dom';
import Home from './Home/Home';

const notFound = () => (<h1>Not found</h1>);
const App = () => (
    <Router>
        <Switch>
            <Route path="/" exact component={Home} />
            <Route component={notFound} />
        </Switch>
    </Router>
);

export default App;
