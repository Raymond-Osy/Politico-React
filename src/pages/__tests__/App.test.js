import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import { render, cleanup } from 'react-testing-library';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import App from '../App';
import { mockStoreData } from '../../__mocks__/mockStore';

const mockStore = configureMockStore([thunk]);
const store = mockStore(mockStoreData);


afterEach(cleanup);

test('<App /> component', () => {
  const app = render(
    <Provider store={store}>
      <Router>
        <App />
      </Router>
    </Provider>
  );
  expect(app).toBeTruthy();
  expect(app).toMatchSnapshot();
});
