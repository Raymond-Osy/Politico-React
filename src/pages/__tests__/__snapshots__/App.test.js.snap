// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="custom-container"
      >
        <nav
          class="row"
        >
          <div
            class="logo-div"
          >
            <a
              href="index.html"
            >
              <h1>
                Politico
              </h1>
            </a>
          </div>
          <div
            class="signup-signin-div"
          >
            <h4>
              <a
                href="signup.html"
              >
                Sign Up
              </a>
              <span
                class="m-side"
              >
                |
              </span>
              <a
                href="signin.html"
              >
                Login
              </a>
            </h4>
          </div>
        </nav>
        <section
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="main-info col-100"
            >
              <h1>
                Welcome to 
                <span
                  class="logo"
                >
                  Politico
                </span>
                , where the Citizens speak!
              </h1>
              <p>
                What more could anyone ask for? It's convinient, free and transparent!
              </p>
              <a
                href="signin.html"
                id="bigRegisterBtn"
              >
                <span>
                  Get Started
                </span>
              </a>
            </div>
          </div>
        </section>
        <footer
          class="footer-container"
        >
          <div
            class="abt-header"
          >
            <h2>
              How Do I Vote?
            </h2>
          </div>
          <div
            class="footer-item-container"
          >
            <div
              class="align-3"
            >
              <div
                class="abt-body"
              >
                <img
                  src="test-file-stub"
                />
                <p>
                  Click the Get Started button
                </p>
              </div>
            </div>
            <div
              class="align-4"
            >
              <div
                class="abt-body"
              >
                <img
                  src="test-file-stub"
                />
                <p>
                  Fill in your credentials in the designated fields
                </p>
              </div>
            </div>
            <div
              class="align-3"
            >
              <div
                class="abt-body"
              >
                <img
                  src="test-file-stub"
                />
                <p>
                  Select & vote your prefered candidate!
                </p>
              </div>
            </div>
          </div>
          <div
            class="footer-container-2"
          >
            <p>
              Copyright © 2018, Raymond Akalonu.
            </p>
          </div>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="custom-container"
    >
      <nav
        class="row"
      >
        <div
          class="logo-div"
        >
          <a
            href="index.html"
          >
            <h1>
              Politico
            </h1>
          </a>
        </div>
        <div
          class="signup-signin-div"
        >
          <h4>
            <a
              href="signup.html"
            >
              Sign Up
            </a>
            <span
              class="m-side"
            >
              |
            </span>
            <a
              href="signin.html"
            >
              Login
            </a>
          </h4>
        </div>
      </nav>
      <section
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="main-info col-100"
          >
            <h1>
              Welcome to 
              <span
                class="logo"
              >
                Politico
              </span>
              , where the Citizens speak!
            </h1>
            <p>
              What more could anyone ask for? It's convinient, free and transparent!
            </p>
            <a
              href="signin.html"
              id="bigRegisterBtn"
            >
              <span>
                Get Started
              </span>
            </a>
          </div>
        </div>
      </section>
      <footer
        class="footer-container"
      >
        <div
          class="abt-header"
        >
          <h2>
            How Do I Vote?
          </h2>
        </div>
        <div
          class="footer-item-container"
        >
          <div
            class="align-3"
          >
            <div
              class="abt-body"
            >
              <img
                src="test-file-stub"
              />
              <p>
                Click the Get Started button
              </p>
            </div>
          </div>
          <div
            class="align-4"
          >
            <div
              class="abt-body"
            >
              <img
                src="test-file-stub"
              />
              <p>
                Fill in your credentials in the designated fields
              </p>
            </div>
          </div>
          <div
            class="align-3"
          >
            <div
              class="abt-body"
            >
              <img
                src="test-file-stub"
              />
              <p>
                Select & vote your prefered candidate!
              </p>
            </div>
          </div>
        </div>
        <div
          class="footer-container-2"
        >
          <p>
            Copyright © 2018, Raymond Akalonu.
          </p>
        </div>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
